File: app/src/main/java/com/example/trash_scan/adapter/UsersAdapter.java
package com.example.trash_scan.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.trash_scan.R;
import com.example.trash_scan.firebase.models.User;
import com.firebase.ui.firestore.FirestoreRecyclerAdapter;
import com.firebase.ui.firestore.FirestoreRecyclerOptions;
import com.squareup.picasso.Picasso;

import de.hdodenhof.circleimageview.CircleImageView;

public class UsersAdapter extends FirestoreRecyclerAdapter<User,UsersAdapter.UsersViewHolder> {

    public interface OnUserClick{
        void onClickUser(int position);
    }
    private Context context;
    private FirestoreRecyclerOptions<User> options;
    private OnUserClick onUserClick;
    public UsersAdapter(Context context,@NonNull FirestoreRecyclerOptions<User> options,OnUserClick onUserClick) {
        super(options);
        this.context = context;
        this.options = options;
        this.onUserClick = onUserClick;
    }

    @Override
    protected void onBindViewHolder(@NonNull UsersViewHolder holder, int position, @NonNull User model) {
        holder.userFullName.setText(model.getUserFirstName() + " " + model.getUserLastName());
        if (!model.getUserProfile().isEmpty()) {
            Picasso.get().load(model.getUserProfile()).into(holder.userProfile);
        }
        holder.itemView.setOnClickListener(v -> onUserClick.onClickUser(position));

    }

    @NonNull
    @Override
    public UsersViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.row_users,parent,false);
        return new UsersViewHolder(view);
    }

    public class UsersViewHolder  extends RecyclerView.ViewHolder {
        private CircleImageView userProfile;
        private TextView userFullName;
        public UsersViewHolder(@NonNull View itemView) {
            super(itemView);
            userFullName = itemView.findViewById(R.id.textUserFullName);
            userProfile  = itemView.findViewById(R.id.imageUserProfile);
        }
    }
}

File: app/src/main/java/com/flysolo/collectorapp/adapters/UsersAdapter.kt
package com.flysolo.collectorapp.adapters

import android.content.Context
import android.text.Layout
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.flysolo.collectorapp.R
import com.flysolo.collectorapp.models.User
import com.squareup.picasso.Picasso
import de.hdodenhof.circleimageview.CircleImageView


class UsersAdapter(val context: Context,val listUsers : List<User>) :
    RecyclerView.Adapter<UsersAdapter.UsersAdapterViewHolder>() {




    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): UsersAdapterViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.row_home_owners,parent,false)
        return UsersAdapterViewHolder(view)
    }

    override fun onBindViewHolder(holder: UsersAdapterViewHolder, position: Int) {
        val user = listUsers[position]
        if (user.userProfile!!.isNotEmpty()) {
            Picasso.get().load(user.userProfile).into(holder.imageProfile)
        }
        holder.textName.text = "${user.userFirstName} ${user.userFirstName}"
        holder.textAddress.text = user.userAddress
        holder.textPhone.text = user.userPhoneNumber
    }

    override fun getItemCount(): Int {
        return listUsers.size
    }
    class UsersAdapterViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView){
        val imageProfile : CircleImageView = itemView.findViewById(R.id.imageProfile)
        val textName :TextView = itemView.findViewById(R.id.textName)
        val textAddress :TextView = itemView.findViewById(R.id.textAddress)
        val textPhone :TextView = itemView.findViewById(R.id.textPhoneNumber)
    }


}